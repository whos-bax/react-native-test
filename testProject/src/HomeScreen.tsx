import React, {useState} from "react";
import {Button, StyleSheet, Text, View} from "react-native";

const HomeScreen = (): React.JSX.Element => {
    const [status, setStatus] = useState("");
    return <View style={styles.viewStyle}>
        <Text testID={"HomeText"}>{status}</Text>
        <Button testID={"HomeButton"} onPress={() => setStatus("Button Pressed")} title={`Status Button`}/>
    </View>
}

const styles = StyleSheet.create({
    viewStyle: {
        backgroundColor: "pink"
    }
})
export default HomeScreen;
