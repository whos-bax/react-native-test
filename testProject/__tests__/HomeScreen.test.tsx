import React from "react";

// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer, {act} from 'react-test-renderer';
import HomeScreen from "../src/HomeScreen.tsx";

const tree =  renderer.create(<HomeScreen/>);

it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
});

it("button Press", () => {
    // press the button
    const button = tree.root.findByProps({testID: "HomeButton"}).props;
    act(() => button.onPress());
    // expect text to equal "Button Pressed"
    const text = tree.root.findByProps({testID: "HomeText"}).props;
    expect(text.children).toEqual("Button Pressed")
})
