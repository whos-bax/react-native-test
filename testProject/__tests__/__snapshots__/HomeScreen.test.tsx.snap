// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "pink",
    }
  }
>
  <Text
    testID="HomeText"
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="HomeButton"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Status Button
      </Text>
    </View>
  </View>
</View>
`;
